client/reader/sidebar/index.jsx

```diff
@@ -254,22 +247,37 @@ const ReaderSidebar = React.createClass( {
 					onTagExists={ this.highlightNewTag }
 					currentTag={ this.state.currentTag } />
 
-				</SidebarRegion>
+			</SidebarRegion>
 
-				{ this.renderAppPromo() }
+			{ this.shouldRenderAppPromo() &&
+				<div className="reader-sidebar__AppPromo">
+					<AppPromo location="reader" locale={ userUtils.getLocaleSlug() } />
+				</div>
+			}
 
 				<SidebarFooter />
 			</Sidebar>
 		);
 	}
 } );
 
+ReaderSidebar.defaultProps = {
+	translate: identity
+};
+
+export { ReaderSidebar };
 export default connect(
 	( state ) => {
 		return {
 			isListsOpen: state.ui.reader.sidebar.isListsOpen,
 			isTagsOpen: state.ui.reader.sidebar.isTagsOpen,
-			subscribedLists: getSubscribedLists( state )
+			subscribedLists: getSubscribedLists( state ),
+			isDesktopPromoDisabled: store.get( 'desktop_promo_disabled' ),
+			isViewportMobile: viewport.isMobile(),
+			isUserLocaleEnglish: 'en' === userUtils.getLocaleSlug(),
+			isChromeOS: isChromeOS(),
+			isDesktopPromoConfiguredToRun: config.isEnabled( 'desktop-promo' ),
+			isUserDesktopAppUser: userSettings.getSetting( 'is_desktop_app_user' )
```

Comment:
…continuing from Slack…

If it were me I'd probably put it in a function right above `mapStateToProps`

```js
…
	render() {
		…
		{ this.props.shouldRenderAd &&
			<AppPromo … />
		}
		…
	}
…

const shouldRenderAd = () => every( [
	! store.get( 'desktop_promo_disabled' ),
	'en' === userUtils.getLocaleSlug(),
	! viewport.isMobile(),
	! isChromeOS(),
	config.isEnabled( 'desktop-promo' ),
	! userSettings.getSetting( 'is_desktop_app_user' )
] );

const mapStateToProps = state => ( {
	…
	shouldRenderAd: shouldRenderAd()
	…
} );
```